Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift

```diff
@@ -48,6 +58,8 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
         self.diagnosticsEngine = diagnosticsEngine
         self.encoder = JSONEncoder.makeWithDefaults()
         self.decoder = JSONDecoder.makeWithDefaults()
+        
+        self.populateTargetTrie()
```

Comment:
I think this should be moved to the search function, such that if the trie is not loaded load it and then return results

Reply:
If we move this to search, we might be inserting the same data to the trie more than once. Imagine between `init` and `searchTargets` collections are added/refreshed, which means FTS and trie are updated also. When `searchTargets` is called, we would attempt to populate the trie using data from FTS, but the trie already has some if not all of the data in the FTS, so we'd be wasting time and resources to insert the same data again. 

And ideally, we'd call populate only once. I think moving this to `searchTargets` would require more work to restrict that.